backend/tests/dining/test_views.py

```diff
@@ -35,6 +37,52 @@ def json(self):
         return Mock(json.load(data), 200)
 
 
+def mock_request_raise_error(*args, **kwargs):
+    raise ConnectionError
+
+
+def mock_request_post_error(*args, **kwargs):
+    class Mock:
+        def json(self):
+            return {"error": None}
+
+    return Mock()
+
+
+class TestTokenAndRequest(TestCase):
```

Comment:
nit: you can override the `setUp` function in `TestCase` so that you don't have to re-define `DiningAPIWrapper()` for every function call. This `setUp` function will be re-invoked for every `test_*` function you have so each test will be independent of each other.